src/Symfony/Bridge/Twig/AppVariable.php

```diff
@@ -145,4 +145,37 @@ public function getDebug()
 
         return $this->debug;
     }
+
+    /**
+     * Returns some or all the existing flash messages:
+     *  * getFlashes() returns all the flash messages
+     *  * getFlashes('notice') returns a simple array with flash messages of that type
+     *  * getFlashes(array('notice', 'error')) returns a nested array of type => messages
+     *
+     * @return array
+     */
+    public function getFlashes($types = null)
+    {
+        // needed to avoid starting the session automatically when looking for flash messages
+        try {
+            $session = $this->getSession();
+            if (null !== $session && !$session->isStarted()) {
+                return array();
+            }
+        } catch (\RuntimeException $e) {
+            return array();
+        }
+
+        if (null === $types || empty($types)) {
+            return $session->getFlashBag()->all();
+        }
+
+        $types = (array) $types;
```

Comment:
Again.. offset 0 may not exist ;-) also the API is still inconsistent with `flashes(['single'])` vs. `flashes('single')`. Both return flat, not just the latter. Like @stof mentioned.. this is too magic.

What about
```php
if (null === $types || array() === $types) {
    return $session->getFlashBag()->all();
}

if (!is_array($types)) {
    return $session->getFlashBag()->get($types);
}

// intersect
```

The `empty` check is too magic (for me) as well; passing `"0", "", etc.` should be passed to `get()` IMO.